envoy/upstream/cluster_manager.h

```diff
@@ -367,6 +367,14 @@ class ClusterManager {
    */
   virtual Config::GrpcMuxSharedPtr adsMux() PURE;
 
+  /**
+   * Replaces the current ADS mux with a new one based on the given config.
+   * Assumes that the given ads_config is yntactically valid (according to the PGV constraints).
+   * @param ads_config an ADS config source to use.
+   * @return the status of the operation.
+   */
+  virtual absl::Status replaceAds(const envoy::config::core::v3::ApiConfigSource& ads_config) PURE;
```

Comment:
wdyt about naming the method `replaceAdxMux`? (sounds more clear to me)

Reply:
Agreed - renamed.